### DISCUSSION TITLE
Is nursery enabled with `select = ["ALL"]`?

### DISCUSSION THREAD

**Discussion by jamesbraza** [astral-sh/ruff#6495] [Q&A] [answered]
Is nursery enabled with `select = ["ALL"]`?

Reading through the docs here, I don't quite see anything on how to enable the nursery: https://beta.ruff.rs/docs/configuration/

I see a lot of this sentence in the docs right now:

> This rule is part of the **nursery**, a collection of newer lints that are still under development. As such, it must be enabled by explicitly selecting RULE123.

But it doesn't really say if one can globally opt into the nursery.

**Comment by tjkuson** [comment]
Selecting `ALL` does not opt into the nursery rules, it opts into all the non-nursery rules. You can select `NURSERY` to opt into the nursery rules. #4601

**Reply by jamesbraza** [reply]
Thanks for sharing, didn't know about that flag.

I use `pyproject.toml` config, should it be `select = ["ALL", "NURSERY"]` or just `select = ["NURSERY"]` (and `"ALL"` is implied)?

**Reply by tjkuson** [reply, marked as answer]
Yeah, the `NURSERY` flag is quite well-hidden (perhaps intentionally, as the nursery rules have problems).

I think you would need to use `select = ["ALL", "NURSERY"]` to select all non-nursery and nursery rules.

**Reply by zanieb** [reply]
I think the nursery is not intended to be as hidden as it is, we just haven't had time to write more documentation about how to use it!
